apps/algolia/lib/algolia/update.ex

```diff
@@ -49,35 +96,38 @@ defmodule Algolia.Update do
       body: request
     }
 
-    opts
-    |> Api.post()
-    |> parse_response()
+    :post |> Api.action(opts) |> parse_response()
   end
 
   defp send_update({:error, error}, _, _) do
     {:error, {:json_error, error}}
   end
 
+  defp response_to_objects(body) do
+    {:ok, json} = Jason.decode(body)
+    Enum.map(json["hits"], fn obj -> %{objectID: obj["objectID"]} end)
+  end
+
   @spec parse_response({:ok, HTTPoison.Response.t()} | {:error, HTTPoison.Error.t()}) ::
           success | error
-  defp parse_response({:ok, %HTTPoison.Response{status_code: 200}}), do: :ok
+  defp parse_response({:ok, %HTTPoison.Response{status_code: 200, body: body}}), do: {:ok, body}
   defp parse_response({:ok, %HTTPoison.Response{} = response}), do: {:error, response}
   defp parse_response({:error, %HTTPoison.Error{} = error}), do: {:error, error}
 
   defp build_request_object(data) do
     Poison.encode(%{requests: data})
   end
 
-  @spec build_data_object(Algolia.Object.t()) :: map
-  def build_data_object(data) do
+  @spec build_action_object(Algolia.Object.t(), String.t()) :: map
+  def build_action_object(data, action) do
     %{
-      action: "addObject",
-      body: do_build_data_object(data)
+      action: action,
+      body: data
     }
   end
 
-  @spec do_build_data_object(Algolia.Object.t()) :: map
-  defp do_build_data_object(data) do
+  @spec to_data_object(Algolia.Object.t()) :: map
+  def to_data_object(data) do
```

Comment:
Same as above.

Reply:
I believe the typing is this way on purpose. This function returns an Algolia object. The route algolia object and stop algolia object both have different fields. Since both share a common type of Map, that was used. I don't think this justifies renaming the function, or changing the typing.